app/models/resource_action_workflow.rb

```diff
@@ -38,7 +38,11 @@ def process_request(state)
       result[:request] = generate_request(state, values)
     else
       ra = load_resource_action(values)
-      ra.deliver_to_automate_from_dialog(values, @target, @requester)
+      if ra.resource_type == 'CustomButton' && CustomButton.find(ra.resource_id).options[:open_url]
```

Comment:
This object should not have intimate knowledge of the `CustomButton` model.  I would expect that the `CustomButton` model should be updated to include a `open_url?` method which would look into the `options` hash.

Also, I would prefer not using `resource_type` to identify the object.  It would be better to use `ra.resource.kind_of?(CustomButton)` but in this case I think you just want: `if ra.resource.try(:open_url?)`